Fix an edge case where ``ExceptionInfo._stringify_exception`` could crash ``pytest.raises``.
